### DISCUSSION TITLE
How can I include trajectories?

### DISCUSSION THREAD

**Discussion by EYErbil** [deepmodeling/deepmd-kit#4072] [Q&A]
Hello, I have trained my model, 
![image](https://github.com/user-attachments/assets/7614156f-367a-4b3f-a39e-e7e7bcf1a73c)
but what I realized is, it has problems when I relax a perturbed structure, it basically converges, but in a very stupid way, 
![image](https://github.com/user-attachments/assets/0f8e13fb-a116-44aa-82aa-164a71e07c4f)
I have trained my model with single point snapshots, meaning that it doesnt know how to proceed to next geometry actually, 
do you think I can include relaxation trajectories of my DFT calculations from before? would that help?

**Comment by wanghan-iapcm** [comment]
Yes, including DFT relaxations will help the generalization in relaxation tasks. 

A better way is to use the concurrent learning that only collects and labels (by DFT) critical configurations along the relaxation trajectories. please refer to https://journals.aps.org/prb/abstract/10.1103/PhysRevB.109.094117 for more details

**Reply by EYErbil** [reply]
but then, how do I feed the algorithm trajectories? The thing I do is convert my data with cp2kdata, and start training, how do I include trajectory files?

**Reply by wanghan-iapcm** [reply]
We usually use [`dpdata`](https://github.com/deepmodeling/dpdata), loading and reading cp2k files is supported by cp2kdata plugin. 
https://docs.deepmodeling.com/projects/dpdata/en/stable/_modules/dpdata/plugins/cp2k.html
https://robinzyb.github.io/cp2kdata/intro.html

**Reply by EYErbil** [reply]
Oh, then the calculations that have trajectories, should have those in the output file of cp2k, right? I dont need to do anything extra, converting the out file with cp2kdata plugin would be enough for the model to understand the trajectories

**Reply by wanghan-iapcm** [reply]
The trajectories are supposed to be convert to format "deepmd/npy" that can be understood by the deepmd-kit. 
you may find examples in the tutorial https://docs.deepmodeling.com/projects/deepmd/en/r2/getting-started/quick_start.html#data-preparation

**Reply by EYErbil** [reply]
Yes, but then, the trajectory files created by cp2k, how do I convert them to NPY,  I mean, trajectory files are only .xyz, they dont have any energy attached to them, only positions, or do I do this wih only the output files of relaxation procedures in cp2k?
            os.system(f"dpdata {output_file} -i cp2k/output -o deepmd/npy -O {data_dir}")
                       system = dpdata.LabeledSystem(file_name=data_dir, fmt='deepmd/npy')

**Reply by EYErbil** [reply]
In this code, I convert .out files to .npy, as labeled systems, and append to multi systems, all output files are single point, so that means no trajectory is present,

**Reply by wanghan-iapcm** [reply]
What I mean is that every "single point" on your optimization trajectory, with labels, can be used for training data.
